internal/fs/rename_go1.7.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build !windows
+// +build !go1.8
+
+package fs
+
+import (
+	"os"
+	"syscall"
+
+	"github.com/pkg/errors"
+)
+
+func rename(src, dst string) error {
+	fi, err := os.Stat(src)
+	if err != nil {
+		return errors.Wrapf(err, "cannot stat %s", src)
+	}
+
+	if dstfi, err := os.Stat(dst); fi.IsDir() && err == nil && dstfi.IsDir() {
```

Comment:
It would be good to add a small note explaining what's going on here. Something like this:
```
// In go 1.8, the behavior of os.Rename changed on non-Windows platforms. It no
// longer allows renames that would replace an existing directory. This has
// always been the case on Windows, though.
// 
// For consistency, we replicate the go1.8 behavior in earlier go versions here.
```